Fix build on macOS

smarsching opened the pull request:
This changeset consists of three changes:

The order of rules in configure/Makefile is changed. For some reason, this is needed in order for `configure/os/CONFIG_PY3.7.Common.darwin-x86` to be generated correctly.

The declaration of the `pyInitLastState` global variable is moved from `pydevsup.h` to `setup.c`. This is needed to avoid a "duplicate symbols" error when linking. As this variable is only used in `setup.c` anyway, there is no downside in moving its declaration there.

The log example application is now only built on Linux. As it depends on the inotify API, which is Linux-specific, it cannot be reasonably built on other platforms. As it is not needed for the pyDevSup to work, it is OK to built it without this application.

mdavidsaver commented:
> The order of rules in configure/Makefile is changed.

That's curious.  What version of GNUMake do you have?

This change make `os/CONFIG_PY$(PY_VER).Common.$(T_A)` the default rule in `configure/` instead of `all`.  Not a huge problem, but something I can see causing confusion down the road.

smarsching commented:
The output of `make --version` is:

```
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

I share your concern regarding the default rule (I did not think about this problem when making the change). I will have a look whether I can find a different workaround.

smarsching commented:
I just reverted the change to the `Makefile` and still everything builds fine (I started from a fresh copy to ensure that `configure/os/CONFIG_PY3.7.Common.darwin-x86` was generated correctly).

I have no clue, why I had this problem originally (maybe I did something else wrong and it was just by accident that I did this right after also changing the order of the rules).

Anyway, I just removed that change from my branch so that now they only contain the two other changes, which certainly make sense.

mdavidsaver commented:
Thanks.
